x/validator-vesting/internal/keeper/querier.go

```diff
@@ -0,0 +1,62 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	supplyexported "github.com/cosmos/cosmos-sdk/x/supply/exported"
+	"github.com/kava-labs/kava/x/validator-vesting/internal/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// NewQuerier returns a new querier function
+func NewQuerier(keeper Keeper) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err sdk.Error) {
+		switch path[0] {
+		case types.QueryCirculatingSupply:
+			return queryGetCirculatingSupply(ctx, req, keeper)
+		default:
+			return nil, sdk.ErrUnknownRequest("unknown cdp query endpoint")
+		}
+	}
+}
+
+func queryGetCirculatingSupply(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
+	circulatingSupply := sdk.ZeroInt()
+	keeper.ak.IterateAccounts(ctx,
+		func(acc authexported.Account) (stop bool) {
+			// exclude module account
+			_, ok := acc.(supplyexported.ModuleAccountI)
+			if ok {
+				return false
+			}
+
+			// periodic vesting account
+			vacc, ok := acc.(vesting.PeriodicVestingAccount)
+			if ok {
+				balance := vacc.GetCoins().AmountOf("ukava")
+				if balance.IsZero() {
+					return false
+				}
+				spendableBalance := vacc.SpendableCoins(ctx.BlockTime()).AmountOf("ukava")
+				circulatingSupply = circulatingSupply.Add(sdk.MinInt(balance, spendableBalance))
```

Comment:
What's the reason for calling `MinInt`?
Is there a situation when `SpendableCoins` are greater than an account's coins?

Reply:
Indeed yes. Spendable coins is a badly named method from the vesting account interface that in fact returns the theoretical spendable balance based the progress of vesting at the current block time. If some of those coins have already been spent, the spendable coins will be greater than the balance.